blocks/bricks/recurrent.py

```diff
@@ -630,3 +631,281 @@ def apply(self, *args, **kwargs):
+    is forked and then added to all the elements appearing in the
+    `sequences` of the next layer (except for "mask".) By default this
+    fork has a bias. This allows direct usage of this class with input
+    supplied only to the first layer. However, if you supply your own
+    inputs to layers above the bottom layer you should take this bias into
+    account or you can replace the fork_protoype to use Linear without
+    bias.
+
+    Parameters
+    ----------
+    transitions : list
+        List of recurrent units to use in each layer.
+        Note: A suffix with layer number is added to transitions' names.
+    fork_prototype : :class:`.FeedForward`, optional
+        A prototype for the  transformation applied to states_name from
+        the states of each layer. The transformation is used when state
+        is forked to the sequences of the next layer. By default a
+         :class:`.Linear` transformation with biases.
+    states_name : string
+        In a stack of RNN the state of each layer is used as input to the
+        next. The `states_name` identify the element of the states of each
+        layer that should be used for this task. By default the element is
+        called "states". To be more precise, this is the name of the
+        element in the outputs of the apply method of each transition
+        (layer) that is used, via fork, as the sequences (input) of the
+        next layer. The same element should also appear in the states
+        parameter of the apply method.
+    fast : bool
+        Use the fast, but also memory consuming, implementation of this
+        code. By default true.
+    skip_connections : bool
+        By default False. When true, each element in the
+        sequences to the bottom layer is passed through a linear
+        transformation, without bias, to each of the sequences of the
+        other layers. When true, all layers must have the same names in
+        their sequences and in the same order.
+
+    Notes
+    -----
+    See :class:`.BaseRecurrent` for more initialization parameters.
+
+    """
+    def suffix(self, name, level):
+        if name == "mask":
+            return "mask"
+        if level == 0:
+            return name
+        if isinstance(name, list):
+            return [self.suffix(n, level) for n in name if n != "mask"]
+        return name + '_' + str(level)
+
+    def __init__(self, transitions, fork_prototype=None, states_name="states",
+                 fast=True, skip_connections=False, **kwargs):
+        super(RecurrentStack, self).__init__(**kwargs)
+
+        self.states_name = states_name
+
+        for level, transition in enumerate(transitions):
+            transition.name += '_' + str(level)
+        self.transitions = transitions
+
+        if fork_prototype is None:
+            # By default use Linear (with bias) as a default fork.
+            fork_prototype = Linear(use_bias=True)
```

Comment:
If we write here

```python
fork_prototype = Linear(use_bias=not skip_connections)
```

then I think all the `skip_connections` bricks will be not necessary anymore. Everything can be done by an external fork _with_ biases.